docs/shared/CONTRIBUTING.md

```diff
@@ -0,0 +1,57 @@
+We have [good first issues][good-first-issue] for new contributors and [help
+wanted][help-wanted] issues for our other contributors.
+
+- `good first issue` has extra information to help you make your first contribution.
+- `help wanted` are issues suitable for someone who isn't a core maintainer.
+
+Maintainers will do their best to regularly make new issues for you to solve and
+then help out as you work on them. 💖
+
+# Philosophy
+
+PRs are most welcome!
+
+- If there isn't an issue for your PR, please make an issue first and explain
+  the problem or motivation for the change you are proposing. When the solution
+  isn't straightforward, for example "Implement missing command X", then also
+  outline your proposed solution. Your PR will go smoother if the solution is
+  agreed upon before you spend a lot of time implementing it.
+  - It's OK to submit a PR directly for problems such as misspellings or other
+    things where the motivation/problem is unambiguous.
+- If you aren't sure about your solution yet, put WIP in the title or open as a
```

Comment:
nit: Maybe just tell people to use draft PRs. These are safer as the merge button isn't clickable.

Reply:
I've been telling people about both because they often open a regular PR by accident, or they think they have it ready then change their mind, and aren't sure how to say "oops I need more time".